middle_end/flambda2/numbers/targetint_31_63.ml

```diff
@@ -14,280 +14,149 @@
 
-    let zero = 0L
+  let hash = hash
 
-    let one = 1L
+  let minus_one = -1L
 
-    let ten = 10L
+  let zero = 0L
 
-    let hex_ff = 0xffL
+  let one = 1L
 
-    let min_value = Int64.min_int
+  let ten = 10L
 
-    let max_value = Int64.max_int
+  let hex_ff = 0xffL
 
-    let sub = Int64.sub
+  let bool_true = one
 
-    let neg = Int64.neg
+  let bool_false = zero
 
-    let shift_left = Int64.shift_left
+  let bool b = if b then bool_true else bool_false
 
-    let shift_right = Int64.shift_right
+  let min_value = Int64.min_int
 
-    let shift_right_logical = Int64.shift_right_logical
+  let max_value = Int64.max_int
 
-    let xor = Int64.logxor
+  let print ppf t = Format.fprintf ppf "%a" print t
 
-    let or_ = Int64.logor
+  let bottom_byte_to_int t = Int64.to_int (Int64.logand t hex_ff)
 
-    let and_ = Int64.logand
+  let sub = Int64.sub
 
-    let mod_ = Int64.rem
+  let neg = Int64.neg
 
-    let div = Int64.div
+  let shift_left = Int64.shift_left
 
-    let mul = Int64.mul
+  let shift_right = Int64.shift_right
 
-    let add = Int64.add
+  let shift_right_logical = Int64.shift_right_logical
 
-    let bottom_byte_to_int t = Int64.to_int (Int64.logand t hex_ff)
+  let xor = Int64.logxor
 
-    let of_char c = Int64.of_int (Char.code c)
+  let or_ = Int64.logor
 
-    let of_int = Int64.of_int
+  let and_ = Int64.logand
 
-    let to_int = Int64.to_int
+  let mod_ = Int64.rem
 
-    let of_int_option i = Some (of_int i)
+  let div = Int64.div
 
-    let of_int32 t = Int64.of_int32 t
+  let mul = Int64.mul
 
-    let to_int32 t = Int64.to_int32 t
+  let add = Int64.add
 
-    let of_int64 t = t
+  let of_char c = Int64.of_int (Char.code c)
 
-    let to_int64 t = t
+  let of_int = Int64.of_int
 
-    let of_float = Int64.of_float
+  let to_int = Int64.to_int
 
-    let to_float = Int64.to_float
+  let of_int_option i = Some (of_int i)
 
-    let to_targetint = Targetint_32_64.of_int64
+  let of_int32 t = Int64.of_int32 t
 
-    let of_targetint = Targetint_32_64.to_int64
+  let to_int32 t = Int64.to_int32 t
 
-    let max_array_length = Int64.sub (Int64.shift_left 1L 54) 1L
+  let of_int64 t = t
 
-    let max_string_length = Int64.sub (Int64.mul 8L max_array_length) 1L
+  let to_int64 t = t
 
-    let max t1 t2 = if Int64.compare t1 t2 < 0 then t2 else t1
+  let of_float = Int64.of_float
 
-    let min t1 t2 = if Int64.compare t1 t2 < 0 then t1 else t2
+  let to_float = Int64.to_float
 
-    let ( <= ) t1 t2 = Stdlib.( <= ) (Int64.compare t1 t2) 0
+  let to_targetint = Targetint_32_64.of_int64
 
-    let ( < ) t1 t2 = Stdlib.( < ) (Int64.compare t1 t2) 0
+  let of_targetint = Targetint_32_64.to_int64
 
-    let to_int_option t =
-      let min_int_as_int64 = Int64.of_int Stdlib.min_int in
-      let max_int_as_int64 = Int64.of_int Stdlib.max_int in
-      if min_int_as_int64 <= t && t <= max_int_as_int64
-      then Some (to_int t)
-      else None
+  let max_array_length = Int64.sub (Int64.shift_left 1L 54) 1L
```

Comment:
This seems quite fragile, and would deserve, at the very least a comment stating this comes from the layout of block headers (and some cross-reference in the definition of headers, to ensure that if/when the headers are modified, then this place is modified as well).

Reply:
Luckily these aren't used any more!